sourmash/commands.py

```diff
@@ -718,10 +719,15 @@ def multigather(args):
         more_files = sourmash_args.load_file_list_of_signatures(args.query_from_file)
         inp_files.extend(more_files)
 
-    query = sourmash_args.load_query_signature(inp_files[0],
-                                               ksize=args.ksize,
-                                               select_moltype=moltype)
-    # set up the search databases
+    # need a "first query"
```

Comment:
ok, let me acknowledge this is tricky/ugly to deal with!

we need to pick a first query so that we can get ksize/moltype. (There may be better ways to do this down the road, but for now, let's stick with the ugly code we have.)

BUT we don't want to rely on file names, and we can actually do everything with `load_file_as_signatures`!

I think something like,

```
query = next(iter(load_file_as_signatures(inp_files[0], ...)))
```
should work to pick off the very first signature, no matter the file type - no need for the if/else.

Reply:
perfect, thanks -- knew we didn't want this hacky implementation, just didn't know the best way to write it!